p2p/peer.go

```diff
@@ -117,8 +117,8 @@ type Peer struct {
 	// events receives message send / receive events if set
 	events *event.Feed
 
-	StaticNodePurposes  *PurposeFlag
-	TrustedNodePurposes *PurposeFlag
+	purposeMu sync.Mutex
+	purpose   PurposeFlag
```

Comment:
any reason for not keeping this plural here and in some other places?

Reply:
Yeah, i was hesitant about names. The type is singular `PurposeFlag` but in reality "holds" many flags....
then the singular flags like `ValidatorPurpose` are of type `PurposeFlag` (singular) which looks ok.

I'll change the name here for now